src/main/resources/GoogleApps/GoogleAppsConfigSheet.xml

```diff
@@ -43,13 +42,92 @@
+$xwiki.ssx.use('GoogleApps.GoogleAppsConfigClass')
 
-#set($class = $doc.getObject('GoogleApps.GoogleAppsConfigClass').xWikiClass)
-#foreach($prop in $class.properties)
-  ; $prop.prettyName
-  : $doc.display($prop.getName())
+#set($configClassName="GoogleApps.GoogleAppsConfig")
+#set($formId = "${section.toLowerCase()}_${configClassName}")
+#set($configDoc = $xwiki.getDocument($configClassName))
+#set($className="GoogleApps.GoogleAppsConfigClass")
+#set($prefix="${configDoc.fullName}_${className}_0")
+#set($obj = $configDoc.getObject($className))
+#macro(t $n)##
+    $services.localization.render("GoogleApps.GoogleAppsConfigClass_${n}")##
+#end
+#macro(tp $n $p)##
+    $services.localization.render("GoogleApps.GoogleAppsConfigClass_${n}",$p)##
 #end
-{{/velocity}}</content>
+
+{{html wiki=true}}
+&lt;form id="$formId" method="post" action="$xwiki.getURL($configDoc, 'saveandcontinue')" onsubmit="cancelCancelEdit()" class="xform"&gt;
+    ; $configDoc.display("activate", 'edit', $obj) &lt;label for="${prefix}_activate"&gt;#t("activate")&lt;/label&gt;
+
+    &lt;fieldset&gt;
+        &lt;legend&gt;#t("communicate")&lt;/legend&gt;
+
+        ; &lt;span class="xHint"&gt;##
+           #tp("communicate.hint",  ['[[', '&gt;&gt;https://accounts.google.com/ManageDomains]]'])&lt;br&gt;
+           #tp("communicate.hint2", ['[[', '&gt;&gt;https://store.xwiki.com/xwiki/bin/view/Extension/GoogleAppsIntegration#installation]]'])&lt;/span&gt;
+
+
+        ; &lt;label for="${prefix}_clientid"&gt;#t("clientid")&lt;/label&gt; ##
+        &lt;span class="xHint"&gt;#t("clientid.hint")&lt;/span&gt;
+        : $configDoc.display("clientid", 'edit', $obj)
+
+        ; &lt;label for="${prefix}_secret"&gt;#t("secret")&lt;/label&gt; ##
+        &lt;span class="xHint"&gt;#t("secret.hint")&lt;/span&gt;
+        : $configDoc.display("secret", 'edit', $obj)
+
+        ## checkboxes for scope, needs JS
+        ; &lt;label&gt;#t("scope")&lt;/label&gt; ##
+        &lt;span class="xHint"&gt;#t("scope.hint")&lt;/span&gt;
+        : &lt;label title="#t('checkbox_mandatory')"&gt;&lt;input type="checkbox" name="scope_identity"  disabled="true" checked&gt;#t("scope_identity")&lt;/label&gt; ##
+          &lt;label title="#t('checkbox_mandatory')"&gt;&lt;input type="checkbox" name="scope_email"     disabled="true" checked&gt;#t("scope_email")&lt;/label&gt; ##
+          &lt;label title="#t('checkbox_mandatory')"&gt;&lt;input type="checkbox" name="scope_gdocs"    disabled="true" checked&gt;#t("scope_gdoc")&lt;/label&gt; ##
+          &lt;label title="#t('checkbox_notyetdone')"&gt;&lt;input type="checkbox" name="scope_avatar"   disabled="true"        &gt;#t("scope_readAvatar")&lt;/label&gt; ##
+          &lt;input type="hidden" name="${prefix}_scope"&gt;
+
+        ; &lt;label for="${prefix}_appname"&gt;#t("appname")&lt;/label&gt; ##
+        &lt;span class="xHint"&gt;#t("appname.hint")&lt;/span&gt;
+        : $configDoc.display("appname", 'edit', $obj)
+
```

Comment:
This is another brake line we can remove and I don't think having it would improve readability, WDYT?

Reply:
Oh, now I understand what you mean with a brake line. 
Well, if I don't put the `##` at the end of the line, the xwiki syntax sees things with a line break and space after that whereas in the case of line 90, I only want a line break. Avoiding the line break would work but `&lt;span` would need to start at the beginning of the line for the span to not appear shifted. Or?
paul

Reply:
I cleaned this up.
paul